xdsl/dialects/riscv.py

```diff
@@ -268,20 +287,25 @@ class RdRsOperation(IRDLOperation, RISCVOp, ABC):
 
     rd: Annotated[OpResult, RegisterType]
     rs: Annotated[Operand, RegisterType]
+    comment: OptOpAttr[StringAttr]
```

Comment:
Not sure how closely you want to integrate comments. To me, comments feel like something that should be more of an attribute that's added after constructing than having them present as part of the IRDL definition of the operation.

Reply:
What API are you thinking of? Here the caller is free not to use the comment arg in the initialiser, and set it after construction, so feels like it would support your use-case

Reply:
You can just add arbitrary attributes to any operation instance, right? There is no need to have them in the IRDL definition, AFAIK.
So essentially, you construct your operation and then do `op.attributes["comment"] = "my comment"`. That would be the other design I could see for this.

Reply:
I would also vote for keeping comments out of the irdl definition, I think

Reply:
But then you lose the type safety

Reply:
Here we know that the attribute is a `StringAttr`, maybe we'll want some `CommentAttr` in the future

Reply:
> But then you lose the type safety

Yes, that's indeed true. I guess you could have an instance check in the comment attribute checking that it is a StringAttr? You have to do some sanity checking anyhow if you were to use it the way I described.

Reply:
I'm still not sure what the disadvantage is to declaring them as optional attributes on the class

Reply:
I am also curious. These attributes are a specific feature of the RISCV dialect that we use to drive the assembly printer. I think keeping them as a local and well-defined concept is more focused, than introducing a project-wide comment concept.